### DISCUSSION TITLE
Transparency & Blurring

### DISCUSSION THREAD

**Discussion by not-nef** [tukaan/tukaan#55] [Question] [answered]
Will tukaan have better transparency options than tkinter and something like acrylic window blurring?

**Comment by rdbende** [comment, marked as answer]
Uhh, good question. Probably yes, I'll make some (Windows-only) blur/acrylic features, but since Tukaan uses the same tk backend as Tkinter, it's possible only with DWMAPI. Tk doesn't have "modern gui features" like transparency, blur, animation or gradients, and it's not planned to get any of these features in the nearby future.
I will play around with DWMAPI on Windows to see what can I do, but as I mentioned, in https://github.com/rdbende/Sun-Valley-ttk-theme/issues/16#issuecomment-1014854466 making the window background acrylic, requires the `-transparentcolor` attribute (so you can click through the window on the empty areas), and it's Windows-only :(

**Reply by rdbende** [reply]
@not-nef It already has it! See #56 
If you've ever tried to maximize and then unmaximize a window that uses BlurWindow, or even simply `-transparentcolor` (like TimerX), you know, that after it, the window gets totally unusable, until you minimize and then unminimize it (by keyboard of course, since the titlebar doesn't work). My blur-ing API doesn't have this annoying behaviour 💪

**Reply by not-nef** [reply]
LETSGOOO @sumeshir26 THIS IS THE BUG I TOLD YOU ABOUT

**Reply by sumeshir26** [reply]
I’ll fix this using rdbemde’s way tomorrow
